### PR TITLE
tests: unit test utils

### PR DESCRIPTION
**Pull request by jacquerie**
The objective of this PR is to have 100% unit test coverage for the files in the `inspirehep/utils` folder.

This entails a complete code review of these files, which will result in just comments: future PRs will address (or delete) these. The only changes that will happen to the code are changes to improve its testability: moving imports in the outer scope, injecting dependencies, and so on.

I'm currently testing too much: a lot of these tests will be deleted, so please don't start reviewing.

### CODE REVIEW

File: inspirehep/utils/arxiv.py [hunk 1]
```diff
@@ -59,8 +59,10 @@ def get_arxiv_id_from_record(record):
             if element.get("value", ""):
                 arxiv_id = element.get("value", "")
 
+    # XXX(jacquerie): should probably be removed.
```
Comment (jalavik) [inline inspirehep/utils/arxiv.py]: See discussion on gitter where we propose to use idutils and some additional checking.

```
In [6]: from idutils import normalize_arxiv

In [7]: normalize_arxiv('oai:arXiv.org:0201.21212')
Out[7]: u'arXiv:oai:arXiv.org:0201.21212'

In [8]: normalize_arxiv('arXiv:0201.21212')
Out[8]: 'arXiv:0201.21212'

In [9]: normalize_arxiv('hep/02011212')
Out[9]: u'arXiv:hep/02011212'

In [10]: normalize_arxiv('0201.21212')
Out[10]: u'arXiv:0201.21212'

In [11]: normalize_arxiv('0201.2121')
Out[11]: u'arXiv:0201.2121'
```
Note missing unicode on one and the **wrong** first case

**Comment by jalavik** [comment]
:+1: Thanks for this!

File: inspirehep/utils/arxiv.py (hunk 1 above)
Comment (jacquerie) [inline inspirehep/utils/arxiv.py]: :+1:

Right, will amend the comment.

File: tests/utils/test_tickets.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+
+class TicketsTests(InvenioTestCase):
+
+    """Tests for the RT ticketing system."""
+
+    @mock.patch('inspirehep.utils.tickets.cfg', rt_cfg)
+    @mock.patch('inspirehep.utils.tickets.rt.Rt')
+    def test_get_instance(self, Rt):
```
Comment (jacquerie) [inline tests/utils/test_tickets.py]: A bad test, because it's a change-detector test: http://googletesting.blogspot.ch/2015/01/testing-on-toilet-change-detector-tests.html.

**Comment by jacquerie** [comment]
This PR hit an interesting milestone: `inspirehep/utils/bibtex.py` has now ~100% test coverage, which ups the total coverage to ~58%. The tests act as some sort of spec for Bibtex exporting, and cover lots of empty and corner cases. In fact, they also uncover a few bugs and some dead code in the current implementation.

The same treatment will be applied to the other exporters (`inspirehep/utils/latex.py`, `inspirehep/utils/cv_latex.py`, and `inspirehep/utils/cv_latex_html.py`).

**Comment by jacquerie** [comment]
- [ ] `inspirehep/utils/bibtex.py`
- [ ] `inspirehep/utils/latex.py`
- [ ] `inspirehep/utils/cv_latex.py`
- [ ] `inspirehep/utils/cvlatex_html_text.py`

**Comment by jalavik** [comment]
@jacquerie :+1: Awesome, perhaps we should already think about merging the current progress as it is already touching many files and may conflict with any ongoing work or Invenio 3 porting.

**Comment by jacquerie** [comment]
Fine by me. Actually, let me have one final review, since I will probably remove lots of tests that are too far-reaching.

**Comment by jacquerie** [comment]
@jalavik Build is green, I had just a minor issue related to title extraction in two tests. Good to go for me.

**Comment by jalavik** [comment]
So now the formatters are getting redone with Invenio 3, including bibtex etc. in #1024 However, no tests are added yet. Shall this branch take care or be merged with #1024 for those files related to bibtex etc. @jacquerie @ioannistsanaktsidis

**Comment by ioannistsanaktsidis** [comment]
@jalavik  I have discussed this with @jacquerie. He will take care of the tests and when my task is finished I will cherry pick his commit on top of mine PR ;) . Sorry for didn't commenting this earlier.

**Comment by jalavik** [comment]
@ioannistsanaktsidis Aha, great! Thanks for the clarification.

**Comment by jacquerie** [comment]
Yeah, this PR is open just for reference, but the testing goodness is still being added.

**Comment by jacquerie** [comment]
Closing because this is not WIP, I'm just using this to salvage code for #1024.
